File: dao_generator/src/org/malopes/generator/Token.java
package org.malopes.generator;

public class Token {

	private String imagem;
	private Classe classe;
	private int indiceTabSimb;
	private int linha;
	private int coluna;

	public Token() {
	}

	public Token(String imagem, Classe classe, int indiceTabSimb, int linha, int coluna) {
		super();
		this.imagem = imagem;
		this.classe = classe;
		this.indiceTabSimb = indiceTabSimb;
		this.linha = linha;
		this.coluna = coluna;
	}

	public String getImagem() {
		return imagem;
	}

	public void setImagem(String imagem) {
		this.imagem = imagem;
	}

	public Classe getClasse() {
		return classe;
	}

	public void setClasse(Classe classe) {
		this.classe = classe;
	}

	public int getIndiceTabSimb() {
		return indiceTabSimb;
	}

	public void setIndiceTabSimb(int indiceTabSimb) {
		this.indiceTabSimb = indiceTabSimb;
	}

	public int getLinha() {
		return linha;
	}

	public void setLinha(int linha) {
		this.linha = linha;
	}

	public int getColuna() {
		return coluna;
	}

	public void setColuna(int coluna) {
		this.coluna = coluna;
	}

	@Override
	public String toString() {
		return imagem;
	}
	
	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((imagem == null) ? 0 : imagem.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Token other = (Token) obj;
		if (imagem == null) {
			if (other.imagem != null)
				return false;
		} else if (!imagem.equals(other.imagem))
			return false;
		return true;
	}
	
	

}

File: trunk/sll_script_lisp_latin/SLL_Projeto/SLL/Token.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SLL_Projeto.SLL
{
    class Token
    {
        public string Imagem { get; set; }
        public ClasseToken Classe { get; set; }
        public int Indice { get; set; }
        public int Linha { get; set; }
        public int Coluna { get; set; }

        public Token()
        {
        }

        public Token(String imagem, ClasseToken classe, int indice, int linha, int coluna)
        {
            this.Imagem = imagem;
            this.Classe = classe;
            this.Indice = indice;
            this.Linha = linha;
            this.Coluna = coluna;
        }

        public override string ToString()
        {
            return Imagem + " " + Classe + " " + Indice + " " + Linha;
        }

        public override bool Equals(object obj)
        {
            if (this == obj)
                return true;
            if (obj == null)
                return false;

            if (GetType() != obj.GetType())
                return false;
            Token other = (Token)obj;
            if (Classe != other.Classe)
                return false;
            if (Imagem != other.Imagem)
                return false;
            return true;
        }

        public override int GetHashCode()
        {
            int prime = 31;
            int result = 1;
            result = prime * result + Classe.GetHashCode();
            result = prime * result + Imagem.GetHashCode();
            return result;
        }
    }
}
